tests/databases/utilities.py

```diff
@@ -0,0 +1,130 @@
+def assert_database_state(state1, state2):
+    assert state1 == state2
+
+
+class HashableDict(dict):
+    def __hash__(self):
+        return hash((frozenset(self), frozenset(self.values())))
+
+    def __eq__(self, other):
+        return super().__eq__(other)
+
+
+def to_hashable(items):
+    for key, value in items:
+        if key == 'a':
```

Comment:
These arbitrary key names need comments.

Reply:
I take your point, but I think its important to note that they are not arbitrary, but they are short versions of the actual names. 'a' = 'atoms', 'b' = 'bonds, 'BB' = building blocks, etc.

Reply:
Obviously, yes, but its still not as clear as it could be with some comments.